lte/gateway/c/session_manager/ChargingGrant.cpp

```diff
@@ -44,22 +45,36 @@ StoredChargingGrant ChargingGrant::marshal() {
   marshaled.service_state                    = service_state;
   marshaled.expiry_time                      = expiry_time;
   marshaled.credit                           = credit.marshal();
+  marshaled.suspended                        = suspended;
   return marshaled;
 }
 
-bool ChargingGrant::is_valid_credit_response(
+CreditValidity ChargingGrant::is_valid_credit_response(
     const CreditUpdateResponse& update) {
-  const uint32_t key           = update.charging_key();
-  const std::string session_id = update.session_id();
+  const uint32_t key             = update.charging_key();
+  const std::string session_id   = update.session_id();
+  CreditValidity credit_validity = VALID_CREDIT;
   if (!update.success()) {
-    MLOG(MERROR) << "Credit update failed RG:" << key
-                 << " code:" << update.result_code() << " for " << session_id;
-    return false;
+    if (DiameterCodeHandler::is_permanent_failure(update.result_code())) {
+      MLOG(MERROR) << "Credit update failed RG:" << key
+                   << " code:" << update.result_code() << " for " << session_id;
+      return INVALID_CREDIT;
+    } else if (DiameterCodeHandler::is_transient_failure(
+                   update.result_code())) {
+      MLOG(MDEBUG) << " Received a transient failure update for RG "
+                   << update.charging_key() << ". Continuing service";
+      credit_validity = TRANSIET_ERROR;
```

Comment:
Why don't we just return TRANSIENT_ERROR here?

Reply:
This is something i still need to figure out: if transient error exist we don't stop the credit fo being installed, but we just suspend it. 

So maybe returning transient error and not checking the validity of the grant would be fine. 

This will be tested on the next PR that will modify the feg